Restructure Flower Datasets docs (#2538)
adap/flower: datasets/doc/source/ref-api-flwr-datasets.rst

flwr\_datasets (Python API reference)
=====================================

> Welcome to the Flower Dataset API Reference Manual! The Flower Dataset library is a pivotal tool designed to address

Reviewer: ```suggestion
Welcome to the Flower Datasets API reference! The Flower Datasets library is a pivotal tool designed to address
```

the unique challenges of managing datasets in the realm of federated learning.

Here, you will find:

> * **Detailed Descriptions**: Understand the core components and functionalities of the Flower Dataset library.

Reviewer: ```suggestion
* **Detailed Descriptions**: Understand the core components and functionalities of the Flower Datasets library.
```

* **Function, Class and Method Definitions**: Get specifications on inputs, outputs, and behaviors for each function class and method available.

> * **Sample Code and Usage**: See the Flower Dataset library in action with short code snippets included in the docstrings.

Reviewer: ```suggestion
* **Sample Code and Usage**: See the Flower Datasets library in action with short code snippets included in the docstrings.
```

For full examples please visit the tutorial section or Flower examples that work end to end using this library.

> .. rubric:: Federated Dataset

Reviewer: This refers to the class name, so let's remove the space:
```suggestion
.. rubric:: FederatedDataset
```
